Map/Include/Field.hpp

```diff
@@ -15,10 +15,20 @@ struct Neighbours
 class Field
```

Comment:
To jest klasa abstrakcyjne. Ma mieć tylko deklaracje metod (patrz linijka 27, 28). Definicje (ciało metody) ma się znajdować w każdej klasie pochodnej (dzieciach klasy Field). To samo dotyczy się pola printability.

Reply:
@marcinwozniak0 Po co przenosić implementacje wspólnych metod, do klas pochodnych?

Reply:
Zdania na ten temat są podzielone. 

Np. takiej klasy abstrakcyjne (bo mamy w niej metody abstrakcyjne = 0)nie możesz stworzyć, a jednak implementujesz jej metody. Możesz wtedy je jedynie testować poprzez klasę dziecko, co nie jest eleganckim rozwiązaniem. 
Jest też kilka spraw związanych z enkapsulacją danych. 
Z drugiej strony, w moim rozwiązaniu duplikujemy kod, co narusza zasadę DRY.

Jak ja bym pisał tą klasę, to bym zrobił to tak, jak poleciłem w komentarzu. Ale możemy wybrać twoje rozwiązanie. Tylko wtedy:
1. bool printability powinno byc private, żeby zachować enkasulacje danych. Dzięki temu klasa dziecko będzie miała dostep do tego pola tylko poprzez gettery/settery klasy bazowej.
2. Zrób clasę FieldTestClass : public Field. I na tej klasie testuj swoje metody.
3. Definicje przenieś do pliku .cpp. w .hpp trzymamy tylko deklaracje.